### PR TITLE
Homework 28.02 clean code

### PR DESCRIPTION
**Pull request by DoTheMonkeyBusiness**

### CODE REVIEW

File: cleancode/src/main/java/com/epam/cleancode/comments/InvalidInputException.java [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package com.epam.cleancode.comments;
 
-public class InvalidInputException extends RuntimeException{
-    String message;
-    public InvalidInputException(String s) {
+public class InvalidInputException extends IllegalArgumentException {
+    private final String message;
+
+    InvalidInputException(String s) {
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/comments/InvalidInputException.java]: Please rename `s` function parameter.

File: cleancode/src/main/java/com/epam/cleancode/comments/InvalidInputException.java [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
         super();
         this.message = s;
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/comments/InvalidInputException.java]: `this` is unnecessary

File: cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java [hunk 3]
```diff
@@ -1,39 +1,31 @@
...
      * @return monthly payment amount
      */
-    public static double calculateMonthlyPayment(
-            int p, int t, double r) {
-
-        //cannot have negative loanAmount, term duration and rate of interest
-        if (p < 0 || t <= 0 || r < 0) {
-            throw new InvalidInputException("Negative values are not allowed");
+    static double calculateMonthlyPayment(
+            int principalAmount, int termOfMortgageInYears, double rateOfInterest) {
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java]: No reason to put function parameters to the next line

File: cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java [hunk 4]
```diff
@@ -1,39 +1,31 @@
...
+        double termInMonths = termOfMortgageInYears * 12.0;
+        double rateInDecimal = rateOfInterest / 100.0;
+        double monthlyRate = rateInDecimal / 12.0;
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java]: `12` can be a constant

File: cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java [hunk 5]
```diff
@@ -1,39 +1,31 @@
...
+
+        if (principalAmount < 0 || termOfMortgageInYears <= 0 || rateOfInterest < 0) {
+            throw new InvalidInputException(INVALID_INPUT_EXCEPTION);
+        }
+        if (rateInDecimal == 0) {
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java:24]: please add empty line before `if`

File: cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java [hunk 6]
```diff
@@ -1,39 +1,31 @@
...
-        //cannot have negative loanAmount, term duration and rate of interest
-        if (p < 0 || t <= 0 || r < 0) {
-            throw new InvalidInputException("Negative values are not allowed");
+    static double calculateMonthlyPayment(
+            int principalAmount, int termOfMortgageInYears, double rateOfInterest) {
+        double termInMonths = termOfMortgageInYears * 12.0;
+        double rateInDecimal = rateOfInterest / 100.0;
+        double monthlyRate = rateInDecimal / 12.0;
+
+        if (principalAmount < 0 || termOfMortgageInYears <= 0 || rateOfInterest < 0) {
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java]: Can be extracted to a separate method, where you can validate function parameters

File: cleancode/src/main/java/com/epam/cleancode/singleresponsibility/Calculator.java [hunk 7]
```diff
@@ -33,12 +46,9 @@ private boolean isNotEmpty(String numbers) {
...
-    }
-
-    private boolean isNotDigits(String numbers) {
-        return !isDigits(numbers);
+        if (numbers == null || !isDigits(numbers)) {
+            throw new WrongFormatException(WRONG_FORMAT_EXCEPTION);
+        }
     }
 
     private boolean isDigits(String numbers) {
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/singleresponsibility/Calculator.java:56]: `"[\\d" + SEPARATOR + "]*"` should be a constant

File: cleancode/src/main/java/com/epam/cleancode/singleresponsibility/WrongFormatException.java [hunk 8]
```diff
@@ -1,4 +1,17 @@
 package com.epam.cleancode.singleresponsibility;
 
-public class WrongFormatException extends RuntimeException {
+public class WrongFormatException extends IllegalArgumentException {
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/singleresponsibility/WrongFormatException.java]: Why you have changed `RuntimeException` to `IllegalArgumentException`?

File: cleancode/src/main/java/com/epam/cleancode/singleresponsibility/WrongFormatException.java [hunk 9]
```diff
@@ -1,4 +1,17 @@
...
+    private final String message;
+
+    WrongFormatException(String s) {
+        super();
+        this.message = s;
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/singleresponsibility/WrongFormatException.java]: The same issue with `this` and `s` parameter

File: cleancode/src/main/java/com/epam/cleancode/singleresponsibility/WrongFormatException.java [hunk 10]
```diff
@@ -1,4 +1,17 @@
...
+
+    }
+
+    @Override
+    public String getMessage() {
+        return message;
+    }
+
```
Comment (egorikftp) [inline cleancode/src/main/java/com/epam/cleancode/singleresponsibility/WrongFormatException.java]: Unnecessary empty line

**Review by DoTheMonkeyBusiness — COMMENTED** [review]
fixed issues

**Pull request closed without merging**